File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"
#include <stdio.h>

unsigned int sum_of_squares(unsigned int number) {
    unsigned int a = 0;
    for(unsigned int i = 1; i <= number; a+=i*i, i+=1);
    return a;
}

unsigned int square_of_sum(unsigned int number){
    unsigned int a = 0;
    for(unsigned int i = 1; i <= number; a+=i, i+=1);
    return a*a;
}

unsigned int number_sum(unsigned int number){
    unsigned int a = 0;
    for(unsigned int i = 1; i <= number; a+=i, i+=1);
    return a;
}

unsigned int difference_of_squares(unsigned int number){
    scanf("%u", &number);
    unsigned int a = 0;
    unsigned int sum = number_sum(number);
    for(unsigned int i = 1; i <= number; a += i*(sum - i), i+=1);
    return a;
    //return square_of_sum(number) - sum_of_squares(number);
}

File: solutions/r/difference-of-squares/1/difference-of-squares.R
# this is a stub function that takes a natural_number
# and should return the difference-of-squares as described
# in the README.md
difference_of_squares <- function(number) {
  if(is.null(number)||number < 0 || number%%1!=0) {
    return(-1)
  }
  sum <- number*(number+1)/2
  square_sum <- sum*sum
  sum_square <- number*(number+1)*(2*number+1)/6
  difference <- square_sum - sum_square
  return(difference)
}